### DISCUSSION TITLE
How do i enable line numbers ?

### DISCUSSION THREAD

**Discussion by xd003** [gsantner/markor#2112] [General] [answered]
Hello, this is my first time using this app. Everything looks pretty damm good but i am unable to find a option to enable the line numbers when i open any file using markor, can anyone please help me out

**Comment by guanglinn** [comment, marked as answer]
You may see the option to enable line numbers in the next version.

![Screenshot_20230922_120008_net.gsantner.markor_test.jpg](https://github.com/gsantner/markor/assets/60415467/6b77e6d0-f819-4149-a109-d4081bae8856)

**Reply by xd003** [reply]
Really glad to see that it has been already added :)

**Reply by xd003** [reply]
Hello @guang-lin i did tested the line number feature by installing the apk compiled from current master branch. It seems to be working fine but each time i open a new file, i need to do these 3 steps to enable line numbers.
1) Tap on the the 3 dot menu on top right
2) Tap on file settings option
3) Tick the line numbers option

Its quite inconvenient for someone who needs line numbers always enabled by default. Is there any global setting to enable line numbers for all files without having to enable it everytime i open a new file ?

**Reply by guanglinn** [reply]
@xd003 Thanks for your feedback on the option to enable line numbers. @gsantner @harshad1

**Reply by xd003** [reply]
Also not just the file number but ideally in order to make this comprehensive. All of these following options in File settings should have a global option which apply on all files by default
1) Wrap Words
2) Line Numbers
3) Syntax Highlighting
4) Auto Format
5) Font Size

Then if a user needs to override these global settings for a specific file, they can very well do that from the 3 dot menu -> File settings which would be only applicable for that file

**Reply by xd003** [reply]
is there any update on this

**Comment by gsantner** [comment]
Read the news.

**Reply by xd003** [reply]
Yeah i did saw the release containing line numbers feature but i was referring to my suggestions here since  its not really that convinient atm - https://github.com/gsantner/markor/discussions/2112#discussioncomment-7087191 & https://github.com/gsantner/markor/discussions/2112#discussioncomment-7087471

**Reply by gsantner** [reply]
Ah, this was not visible in the email.

For sure settings and file specific settings can be improved. But it both needs somebody to think off a clever solution AND (thats the bigger part) somebody to implement it. If you think it would be viable for you to do both feel free to work & develop a concept.

At the end, have as little buttons and toggles around as needed. Keep it simple. Want rather to debloat stuff then to add 20 new buttons.
